kata-deploy/action/test-kata.sh

```diff
@@ -0,0 +1,163 @@
+    # if a node is found which matches node-select, the output will include a column for node name,
+    # NAME. Let's look for that 
+    if [[ -z $(kubectl get nodes --selector katacontainers.io/kata-runtime | grep NAME) ]]
+    then
+        return 0
+    else
+        echo "waiting for kata-runtime label to be removed"
+        sleep "$sleep_time"
+        wait_time=$((wait_time-sleep_time))
+    fi
+  done
+
+  echo "failed to cleanup"
+  return 1
+}
+
+
+function run_test() {
+	PKG_SHA=$1
+	YAMLPATH="https://raw.githubusercontent.com/amshinde/kata-packaging/$PKG_SHA/kata-deploy"
+	echo "verify connectivity with a pod using Kata"
+
+	deployment=""
+	busybox_pod="test-nginx"
+	busybox_image="busybox"
+	cmd="kubectl get pods | grep $busybox_pod | grep Completed"
+	wait_time=120
+	sleep_time=3
+
+    configurations=("nginx-deployment-qemu" "nginx-deployment-qemu-virtiofs")
+    for deployment in "${configurations[@]}"; do
+	  # start the kata pod:
+	  kubectl apply -f "$YAMLPATH/examples/${deployment}.yaml"
+
+	  # in case the control plane is slow, give it a few seconds to accept the yaml, otherwise
+	  # our 'wait' for deployment status will fail to find the deployment at all
+	  sleep 3 
+
+	  kubectl wait --timeout=5m --for=condition=Available deployment/${deployment}
+	  kubectl expose deployment/${deployment}
+
+	  # test pod connectivity:
+	  kubectl run $busybox_pod --restart=Never --image="$busybox_image" -- wget --timeout=5 "$deployment"
+	  waitForProcess "$wait_time" "$sleep_time" "$cmd"
+	  kubectl logs "$busybox_pod" | grep "index.html"
+	  kubectl describe pod "$busybox_pod"
+
+	  # cleanup:
+	  kubectl delete deployment "$deployment"
+	  kubectl delete service "$deployment"
+	  kubectl delete pod "$busybox_pod"
+    done
+}
+
+
+function test_kata() {
+	set -x
+
+    [[ -z "$PKG_SHA" ]] && die "no PKG_SHA provided"
+	echo "$PKG_SHA"
+
+	#kubectl all the things
+	kubectl get pods,nodes --all-namespaces
+
+	YAMLPATH="https://raw.githubusercontent.com/amshinde/kata-packaging/$PKG_SHA/kata-deploy"
+	
+	kubectl apply -f "$YAMLPATH/kata-rbac.yaml"
+
+	# apply runtime classes:
+	kubectl apply -f "$YAMLPATH/k8s-1.14/kata-qemu-runtimeClass.yaml"
+	kubectl apply -f "$YAMLPATH/k8s-1.14/kata-qemu-virtiofs-runtimeClass.yaml"
+
+	kubectl get runtimeclasses
+
+	wget "$YAMLPATH/kata-deploy.yaml"
+	wget "$YAMLPATH/kata-cleanup.yaml"
+
+	# update deployment daemonset to utilize the container under test:
+	sed -i "s#katadocker/kata-deploy#katadocker/kata-deploy-ci:${PKG_SHA}#g" kata-deploy.yaml
+	sed -i "s#katadocker/kata-deploy#katadocker/kata-deploy-ci:${PKG_SHA}#g" kata-cleanup.yaml
+
+	cat kata-deploy.yaml
+
+	# deploy kata:
+	kubectl apply -f kata-deploy.yaml
+
+	# in case the control plane is slow, give it a few seconds to accept the yaml, otherwise
+        # our 'wait' for deployment status will fail to find the deployment at all. If it can't persist
+	# the daemonset to etcd in 30 seconds... then we'll fail.
+	sleep 30
```

Comment:
There must be a better way for checking k8s status. Please tell me there's a better way... :pray:

Reply:
This isn't kubernetes, necessarily, but AKS behavior.  I know...